### PR TITLE
fix(PE-7011): fix target id on spawn

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/utils/transactionUtils/transactionUtils.tsx [hunk 1]
```diff
@@ -694,3 +695,34 @@ export function userHasSufficientBalance<T extends Record<string, number>>({
...
+        ttlSeconds: 3600,
+      },
+    },
+    ...state,
+  };
+}
+
+export function createAntStateForOwner(owner: string, targetId: string) {
+  return createDefaultAntState({
+    owner: owner.toString(),
```
Comment (kunstmusik) [inline src/utils/transactionUtils/transactionUtils.tsx:718]: toString() not necessary here in this function when it's already a string argument

File: src/state/actions/dispatchArIOInteraction.ts [hunk 2]
```diff
@@ -58,8 +58,16 @@ export default async function dispatchArIOInteraction({
...
+            payload.state ||
+            (payload.targetId
+              ? createAntStateForOwner(
+                  owner.toString(),
+                  payload?.targetId?.toString(),
+                )
+              : undefined);
           antProcessId = await spawnANT({
-            state: payload.state,
+            state,
```
Comment (kunstmusik) [inline src/state/actions/dispatchArIOInteraction.ts:70]: Do we ever want to spawn an ant where state may be undefined?

File: src/state/actions/dispatchArIOInteraction.ts [hunk 3]
```diff
@@ -58,8 +58,16 @@ export default async function dispatchArIOInteraction({
         let antProcessId: string = payload.processId;
 
         if (antProcessId === 'atomic') {
+          const state =
+            payload.state ||
+            (payload.targetId
```
Comment (kunstmusik) [inline src/state/actions/dispatchArIOInteraction.ts:63]: Seems odd here to check payload.targetId then later use payload?.targetId?. Can this be condensed down to:

```
 const state =
            payload.state ||
            createAntStateForOwner(
                  owner.toString(),
                  payload?.targetId?.toString(),
                );
                ```

File: src/state/actions/dispatchArIOInteraction.ts (hunk 2 above)
Comment (atticusofsparta) [reply src/state/actions/dispatchArIOInteraction.ts:70]: Yup

File: src/utils/transactionUtils/transactionUtils.tsx [hunk 4]
```diff
@@ -694,3 +695,34 @@ export function userHasSufficientBalance<T extends Record<string, number>>({
...
+        ttlSeconds: 3600,
+      },
+    },
+    ...state,
+  };
+}
+
+export function createAntStateForOwner(owner: string, targetId?: string) {
+  return createDefaultAntState({
+    owner: owner.toString(),
```
Comment (kunstmusik) [inline src/utils/transactionUtils/transactionUtils.tsx:718]: still owner.toString() here instead of just owner

**Review by kunstmusik — APPROVED** [review]

**Pull request merged**